src/Model/Item.php

```diff
@@ -1439,6 +1449,27 @@ public static function insert(array $item, int $notify = 0, bool $post_local = t
 		return $post_user_id;
 	}
 
+	private static function hasRestrictions(array $item, int $author_id, int $restrictions = null): bool
+	{
+		if (empty($restrictions) || ($author_id == $item['author-id'])) {
+			return false;
+		}
+
+		if (($restrictions & self::CANT_REPLY) && ($item['verb'] == Activity::POST)) {
+			return true;
+		}
+
+		if (($restrictions & self::CANT_REPLY) && ($item['verb'] == Activity::ANNOUNCE)) {
```

Comment:
Shouldn't this be `CANT_ANNOUNCE`? 
```suggestion
		if (($restrictions & self::CANT_ANNOUNCE) && ($item['verb'] == Activity::ANNOUNCE)) {
```

Reply:
Good catch!